manager/cve_handler.py

```diff
@@ -152,6 +164,129 @@ def _full_query(rh_account_id, synopsis):
+                        .join(VulnerablePackageCVE, on=(SystemVulnerablePackage.vulnerable_package_id == VulnerablePackageCVE.vulnerable_package_id))
+                        .join(CveMetadata, on=(VulnerablePackageCVE.cve_id == CveMetadata.id))
+                        .where(SystemVulnerablePackage.rh_account_id == rh_account_id)
+                        )
+
+        fixed_subq = (SystemVulnerabilities
+                    .select(
+                            CveMetadata.id.alias("cve_id"),
+                            CveMetadata.cve,
+                            SystemPlatform.id.alias("system_id"),
+                            SystemPlatform.inventory_id,
+                            SystemPlatform.display_name,
+                            SystemPlatform.last_evaluation,
+                            SystemPlatform.advisor_evaluated,
+                            SystemPlatform.last_upload,
+                            SystemPlatform.stale_timestamp,
+                            SystemPlatform.stale_warning_timestamp,
+                            SystemPlatform.culled_timestamp,
+                            InsightsRule.name.alias("rule_id"),
+                            InsightsRule.active.alias("rule_active"),
+                            InsightsRule.description_text,
+                            InsightsRule.reason_text,
+                            InsightsRule.resolution_text,
+                            InsightsRule.kbase_node_id,
+                            InsightsRule.more_info_text,
+                            SystemVulnerabilities.status_text.alias("status_text"),
+                            SystemVulnerabilities.rule_hit_details,
+                            SystemVulnerabilities.when_mitigated,
+                            SystemVulnerabilities.first_reported,
+                            SystemVulnerabilities.mitigation_reason,
+                            SystemVulnerabilities.advisories,
+                            SystemVulnerabilities.status_id,
+                            fn.COALESCE(SystemVulnerabilities.advisory_available, True).alias("advisory_available"),
+                            fn.COALESCE(SystemVulnerabilities.remediation_type_id, remediation.PLAYBOOK.value).alias("remediation_type_id"),
+                            )
+                    .distinct()
+                    .join(SystemPlatform, on=(SystemVulnerabilities.system_id == SystemPlatform.id) &
+                                    system_is_active(rh_account_id=rh_account_id))
+                    .join(InsightsRule, JOIN.LEFT_OUTER, on=(InsightsRule.id == SystemVulnerabilities.rule_id))
+                    .join(CveMetadata, on=(SystemVulnerabilities.cve_id == CveMetadata.id))
+                    .where(SystemVulnerabilities.rh_account_id == rh_account_id)
+                    .where(system_is_vulnerable(rule_subselect=False))
+                    )
+
+        full_subq = (fixed_subq | unfixed_subq)
```

Comment:
the union should be done based on the `advisory_available` filter as described above